### ISSUE TITLE
Operator does not apply ownerReferences on 1Password item update

### ISSUE THREAD

**Issue by codeshrew** [1Password/onepassword-operator#101]
### Your environment

Operator Version: 1password/onepassword-operator:1.4.0

Connect Server Version: 1password/connect-api:1.5.1

Kubernetes Version: v1.21.5-eks-bc4871b

1Password Operator version used:
- https://1password.github.io/connect-helm-charts/
- Chart version: 1.7.1
- App version: 1.5.1
- Chart values used:
```yaml
operator:
  create: true
  version: "1.4.0"
```

## What happened?
When updating a 1Password item (e.g. Secure Note or Login) in the 1Password Mac native application, the onepassword-operator fails to apply the `ownerReferences` values to the `Secret` in the Kubernetes cluster. This causes ArgoCD to believe it owns the `Secret` and it is deleted.

## What did you expect to happen?
When updating a 1Password item in the Mac native application the operator should apply the same `ownerReferences` that a `OnePasswordItem` does on initial creation of the `Secret`.

## Steps to reproduce
1. Add `OnePasswordItem` to git repo for the `example-argocd-project`. Git commit and push.
2. ArgoCD sees the new resource and creates it in the `example` namespace in the cluster
3. 1Password Operator sees the new `OnePasswordItem` in the `example` namespace and creates the `Secret` in the same namespace. This properly includes the `ownerReferences` field and values. It also has the labels from the `OnePasswordItem` referencing the ArgoCD project.
4. Update the 1Password item in the native Mac app (tested with both Secure Notes and Login item types). Tested with several update methods:
	- Adding new label:field pair
	- Removing label:field pair
	- Adding multiple label:field pairs
	- Editing the password field of a Login
5. The 1Password Connect/Operator detects the change and updates the `Secret`. This newly updated `Secret` is now missing the `ownerReferences` field but still has the `labels` from the `OnePasswordItem` indicating that is is part of the ArgoCD project
6. ArgoCD sees the `Secret` without the `ownerReferences` but with the `labels` and deletes the `Secret`

## Notes & Logs

`OnePasswordItem` referenced in step 1 above.

```yaml
apiVersion: onepassword.com/v1
kind: OnePasswordItem
metadata:
  name: test
spec:
  itemPath: "vaults/REDACTED/items/test"
```

`Secret` created during step 3 above. Includes `ownerReferences`.

```yaml
apiVersion: v1
data:
  password: REDACTED
  username: REDACTED
kind: Secret
metadata:
  annotations:
    kubectl.kubernetes.io/last-applied-configuration: |
      {"apiVersion":"onepassword.com/v1","kind":"OnePasswordItem","metadata":{"annotations":{},"labels":{"argocd.argoproj.io/instance":"example-argocd-project","tags.datadoghq.com/env":"dev"},"name":"test","namespace":"example"},"spec":{"itemPath":"vaults/REDACTED/items/test"}}
    operator.1password.io/item-path: vaults/REDACTED/items/REDACTED
    operator.1password.io/item-version: "4"
  creationTimestamp: "2022-04-07T17:35:41Z"
  labels:
    argocd.argoproj.io/instance: example-argocd-project
    tags.datadoghq.com/env: dev
  name: test
  namespace: example
  ownerReferences:
  - apiVersion: onepassword.com/v1
    kind: OnePasswordItem
    name: test
    uid: REDACTED
  resourceVersion: "REDACTED"
  uid: REDACTED
type: Opaque
```

`Secret` updated during step 5 above. Excludes `ownerReferences`.

```yaml
apiVersion: v1
data:
  password: REDACTED
  username: REDACTED
kind: Secret
metadata:
  annotations:
    kubectl.kubernetes.io/last-applied-configuration: |
      {"apiVersion":"onepassword.com/v1","kind":"OnePasswordItem","metadata":{"annotations":{},"labels":{"argocd.argoproj.io/instance":"example-argocd-project","tags.datadoghq.com/env":"dev"},"name":"test","namespace":"example"},"spec":{"itemPath":"vaults/REDACTED/items/test"}}
    operator.1password.io/item-path: vaults/REDACTED/items/REDACTED
    operator.1password.io/item-version: "4"
  creationTimestamp: "2022-04-07T17:35:41Z"
  labels:
    argocd.argoproj.io/instance: example-argocd-project
    tags.datadoghq.com/env: dev
  name: test
  namespace: example
  resourceVersion: "REDACTED"
  uid: REDACTED
type: Opaque
```

**Comment by jpcoenen** [comment]
Thanks for reporting this. I can confirm that the `ownerReference` indeed disappears when updating the item in 1Password.

The root-cause seems to be [this line](https://github.com/1Password/onepassword-operator/blob/main/pkg/onepassword/secret_update_handler.go#L145) where the secret is updated (replaced) without an `ownerReference`.

**Comment by jpcoenen** [comment]
Merged in #103
